### PR TITLE
Setting priority of the ztunnel daemonset

### PR DESCRIPTION
**Pull request by RobCannon**
**Please provide a description of this PR:**
This is to set the priorityClass of the ztunnel Daemonset so that the Kubernetes scheduler when ensure it runs on all nodes (unless there are affinity rules that prohibit ztunnel on certain nodes).   This change matches the existing configuration of the Daemonset for istio-cni-node Daemonset.  That configuration can be found here: https://github.com/ali-research/istio/blob/a2fc718683e7660b51ba336cdb4df46e084277bd/manifests/charts/istio-cni/templates/daemonset.yaml#L62

I have brought up this idea in https://github.com/istio/istio/issues/47867 and no one has objected to the idea.

I updated the helm chart but I also found a manifest at `operator/cmd/mesh/testdata/manifest-generate/output/ztunnel.golden.yaml` that should probably be updated, too.

### CODE REVIEW

File: operator/cmd/mesh/testdata/manifest-generate/output/ztunnel.golden.yaml [hunk 1]
```diff
@@ -220,6 +220,7 @@ spec:
             add:
             - NET_ADMIN
           privileged: true
+      priorityClassName: system-node-critical
```
Comment (hanxiaop) [inline operator/cmd/mesh/testdata/manifest-generate/output/ztunnel.golden.yaml:223]: I think this needs to automatically be updated with a `make gen` command.

**Review by hanxiaop — COMMENTED** [review]
The change LGTM

File: operator/cmd/mesh/testdata/manifest-generate/output/ztunnel.golden.yaml (hunk 1 above)
Comment (RobCannon) [reply operator/cmd/mesh/testdata/manifest-generate/output/ztunnel.golden.yaml]: `make gen` undid that change, so I removed it and squashed the commit.

**Review by howardjohn — COMMENTED** [review]
Only concern: I had thought you cannot use this priorityClass outside of kube-system?

Maybe that is a GKE only thing...

**Comment by RobCannon** [comment]
> Only concern: I had thought you cannot use this priorityClass outside of kube-system?
> 
> Maybe that is a GKE only thing...

I have not heard of that (I am using EKS), but I would point out that the istio-cni-node Daemonset has this `priorityClassName` and it is non-confgurable from the helm chart, so I can't imagine that it would be an issue on GKE.

**Comment by howardjohn** [comment]
In GKE istio-cni is deployed in kube-system specifically for that reason
though

**Comment by linsun** [comment]
> Only concern: I had thought you cannot use this priorityClass outside of kube-system?
> 
> Maybe that is a GKE only thing...

Yeah, I think this maybe GKE only, we noticed it during testings of 3 large clouds of Istio+CNI.

**Review by linsun — APPROVED** [review]

**Comment by linsun** [comment]
@RobCannon thx for the PR, can you fix your CLA?  We need it signed before this PR can be merged.

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**